lib/src/Timer/Timer.tsx

```diff
@@ -6,80 +6,67 @@ export type TimerProps = JSX.IntrinsicElements["div"] & {
+    const startInterval = (timeout: number): void => {
+        setTimer(convertMStoTime(timeout));
+        clearInterval();
+        setInnerInterval(
+            setInterval(() => {
                 if (timeout > 0) {
                     timeout = timeout - 1000;
-                    this.setState({ timer: this.convertMStoTime(timeout) }, () => {
-                        if (timeout === 0) {
-                            this.props.callback();
-                            this.clearInterval();
-                        }
-                    });
+                    setTimer(convertMStoTime(timeout));
+                    if (timeout === 0) {
+                        callback?.();
+                        clearInterval();
+                    }
                 }
-            }, 1000);
-        });
-    }
-
-    clearInterval(): void {
-        if (this.innerInterval) {
-            clearTimeout(this.innerInterval);
-            this.innerInterval = null;
-        }
-    }
-
-    convertMStoTime(value: number): string {
-        const date: Date = new Date(value);
-        return (
-            (date.getUTCHours() > 0 ? date.getUTCHours() + ":" : "") +
-            (date.getUTCMinutes() < 10 ? "0" + date.getUTCMinutes() : date.getUTCMinutes()) +
-            ":" +
-            (date.getUTCSeconds() < 10 ? "0" + date.getUTCSeconds() : date.getUTCSeconds())
+            }, 1000)
         );
-    }
+    };
 
-    componentDidMount() {
-        if (this.props.duration !== null && this.props.duration !== undefined) {
-            this.startInterval(this.props.duration);
+    const clearInnerInterval = (): void => {
+        if (innerInterval) {
+            clearInterval(innerInterval);
+            setInnerInterval(null);
         }
-    }
+    };
 
-    componentWillUnmount() {
-        this.clearInterval();
-    }
+    const convertToTwoDigits = (rawNumber: number): string => `${rawNumber < 10 ? "0" : ""}${rawNumber}`;
 
-    componentDidUpdate(prevProps: TimerProps): void {
-        if (prevProps.duration !== this.props.duration) {
-            if (this.props.duration !== null && this.props.duration !== undefined) {
-                this.startInterval(this.props.duration);
-            }
+    const convertMStoTime = (milliseconds: number): string => {
+        const rawSeconds: number = milliseconds / 1000;
+        const rawMinutes: number = rawSeconds / 60;
+        const displaySeconds: number = rawSeconds % 60;
+        const displayMinutes: number = Math.floor(rawMinutes % 60); // get remainder minutes
+        const displayHours: number = Math.floor(rawMinutes / 60); // get converted hours
+        return (displayHours > 0 ? displayHours + ":" : "") + convertToTwoDigits(displayMinutes) + ":" + convertToTwoDigits(displaySeconds);
+    };
+
+    React.useEffect(() => {
+        if (duration >= 1000) {
+            // only start timer if there's at least 1 second set
+            clearInnerInterval();
```

Comment:
is this step needed? am assuming on re-render/unmount the clean up code would be called?